### PR TITLE
Fix input validation alert

### PR DESCRIPTION
**Pull request by Charles-Gagnon**
Fixes https://github.com/microsoft/azuredatastudio/issues/15748

The scenario that was breaking was this : 

1. InputBox component has custom validation (async)
2. Invalid text is entered into box - validity checks happen and the state is stored as "invalid"
3. Text is changed to be valid
4. Immediately the VS Input Box runs validate() (it does this for every text change). But the state was still cached as invalid and thus it sent out an aria alert for the invalid state
5. Soon after the async validation finished and the inputbox component manually calls validate - this updated the input element so it displayed correctly. But by then it was too late for the alert since that had already gone out. 

Example screen reader text before fix (the second "Error: Invalid config path" text shouldn't be there): 

```
C:\Users\chgagnon\.kube\config unselected
g
Error: Invalid config path 
Kube Config File Path. 
Error loading cluster contexts. ENOENT: no such file or directory, open 'C:\Users\chgagnon\.kube\confi' 
g
Error: Invalid config path 
Loading cluster contexts completed 
Kube Config File Path

```

The fix here is to allow users of the custom SQL Input Box to optionally skip running validate on the base input box except when it's specifically told to - basically giving full manual control to the thing creating the input box to handle running validate. 

Example screen reader text after fix : 

```
Kube Config File Path  edit  C:\Users\chgagnon\.kube\config
g
Error: Invalid config path 
Error loading cluster contexts. ENOENT: no such file or directory, open 'C:\Users\chgagnon\.kube\confi' 
Kube Config File Path. 
g
Loading cluster contexts completed 
Kube Config File Path
```

It's pretty likely that this kind of issue exists for other similar controls, but I'm purposely just keeping this scoped to where I know there's an issue and we can fix up other areas as needed going forward.

### CODE REVIEW

File: src/sql/base/browser/ui/inputBox/inputBox.ts
Comment (alanrenmsft) [inline src/sql/base/browser/ui/inputBox/inputBox.ts:150]: if you merge this in right now, you will need to update the vscode merge pr to add override here, right?

**Review by alanrenmsft — APPROVED** [review]

File: src/sql/base/browser/ui/inputBox/inputBox.ts
Comment (Charles-Gagnon) [reply src/sql/base/browser/ui/inputBox/inputBox.ts:150]: Correct, I plan on waiting until that's merged in before getting this in

**Review by nasc17 — APPROVED** [review]

**Pull request merged**
